### PR TITLE
Update lru dependency 

### LINKED ISSUE #1917: Update `lru` dependency
**Issue summary**

It's required that https://github.com/ChainSafe/forest/pull/1896 will be first merged

**Other information and links**

### PR DESCRIPTION
**Pull request by tyshko5**
**Summary of changes**
Changes introduced in this pull request:
- 

**Reference issue to close (if applicable)**

Closes #1917

**Other information and links**

### CODE REVIEW

File: blockchain/chain/src/lib.rs [hunk 1]
```diff
@@ -1,6 +1,6 @@
 // Copyright 2019-2022 ChainSafe Systems
 // SPDX-License-Identifier: Apache-2.0, MIT
-
+#![feature(const_option)]
```
Comment (lemmih) [inline blockchain/chain/src/lib.rs:3]: We want to be stable so this is not an option.

Comment (lemmih) [reply blockchain/chain/src/lib.rs:3]: You can manually check for `None`, though.

Comment (tyshko5) [reply blockchain/chain/src/lib.rs]: Removed it.

File: blockchain/chain_sync/src/bad_block_cache.rs [hunk 2]
```diff
@@ -1,6 +1,8 @@
 // Copyright 2019-2022 ChainSafe Systems
 // SPDX-License-Identifier: Apache-2.0, MIT
 
+use std::num::NonZeroUsize;
+
```
Comment (LesnyRumcajs) [inline blockchain/chain_sync/src/bad_block_cache.rs:5]:
```suggestion
```

**Review by LesnyRumcajs — APPROVED** [review]

File: blockchain/chain/src/store/chain_store.rs [hunk 3]
```diff
@@ -47,7 +48,7 @@ const BLOCK_VAL_PREFIX: &[u8] = b"block_val/";
 // A cap on the size of the future_sink
 const SINK_CAP: usize = 200;
 
-const DEFAULT_TIPSET_CACHE_SIZE: usize = 8192;
+const DEFAULT_TIPSET_CACHE_SIZE: Option<NonZeroUsize> = NonZeroUsize::new(8192);
```
Comment (lemmih) [inline blockchain/chain/src/store/chain_store.rs:51]: You could write something like:

```suggestion
// XXX: Update when const `option.unwrap()` is stable.
const DEFAULT_TIPSET_CACHE_SIZE: Option<NonZeroUsize> = {
  match NonZeroUsize::new(8192) {
    Some(usize) => usize,
    None => panic!("Why God?! Why?"),
  }};
```

Comment (lemmih) [reply blockchain/chain/src/store/chain_store.rs:51]: Keeping what you have is fine, though.

**Review by lemmih — APPROVED** [review]

**Pull request merged**